elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1712[ELY-1712]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-699[EAP7-699]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Support for RFC 5424 should be enhanced, according to the following parameters, all of which should be added to the ```syslogAuditLogType``` in the Elytron schema
+// Could not find information about this in legacy
+** A ```syslog-format``` attribute should be added which supports a value of ```RFC5424```
+** Some of the parameters needed for RFC 5424 already exist in the Elytron codebase, which are the following:
+*** severity
+*** server address
+*** port
+*** setting TCP
+*** setting SSL
+*** setting socket factory
+*** setting hostname
+** The current attributes available in the Elytron schema, which will be used for RFC 5424, are:
+*** server-address
+*** port
+*** transport type
+*** message format
+*** hostname (?)
+*** ssl-context
+** Additional attributes which will need to be added are:
+*** include timeQuality
+*** include origin
+*** include meta
+*** facility values
+*** priority mapping from facility and severity values, in accordance with https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+** The following attributes will also be available, but should be discarded if timeQuality is not specified:
+*** tzKnown
+*** isSynced
+*** syncAccuracy
+** The following attributes will also be available, but should be discarded if origin is not specified:
+*** ip
+*** enterpriseId
+*** software
+*** swVersion
+** The following attributes will also be available, but should be discarded if meta is not specified:
+*** sequenceId
+*** sysUpTime
+*** language
+
+* Support for RFC 3164 should be added, based off of the implementation for RFC 5424 above, as such should also be added to the ```syslogAuditLogType``` in the Elytron schema
+** The ```syslog-format``` attribute described in the RFC 5424 section above should also support a value of ```RFC3164```
+** All parts of RFC 3164 are included in the RFC 5424 specification, as such the support for RFC 3164 should be only
+changing the message format and not require any additional implementation
+** The implementation of the RFC 3164 message format can be implemented based on the https://github.com/kabir/tamper-detecting-audit-log/blob/master/src/main/java/org/jboss/audit/log/TcpSyslogHandler.java[tamper-detecting-audit-log]
+
+* Support for user formatted strings should be added
+** Currently, this feature does not have a way to be implemented
```

Comment:
Will be this somehow improved? Can you please elaborate?

Reply:
This is something that was requested in the RFC, but I'm unsure what it would refer to exactly. I included it in the proposal for feedback on what should be added for this capability

Reply:
What I am looking for is description how can user benefit from this new "feature". IMHO it doesn't matter how it will be implemented, rather than how the user can benefit from it.

Reply:
I will discuss this with support and @darranl